pipelines_version_insertion.py

```diff
@@ -0,0 +1,113 @@
+import os
+import json
+
+from azure.storage.blob import BlobServiceClient
+from azure.core.exceptions import ResourceExistsError
+
+from sys import argv
+from cryptography.fernet import Fernet
+from dotenv import load_dotenv
+from custom_exceptions import ConnectionStringError
+
+load_dotenv()
+
+KEY = os.getenv("NACHET_BLOB_PIPELINE_DECRYPTION_KEY")
+BLOB_STORAGE_ACCOUNT_NAME = os.getenv("NACHET_BLOB_PIPELINE_NAME")
+CONNECTION_STRING = os.getenv("NACHET_AZURE_STORAGE_CONNECTION_STRING")
+
+
+class PipelineInsertionError(Exception):
+    pass
+
+
+def insert_new_version_pipeline(
+        pipelines_json: dict,
+        connection_string: str,
+        pipeline_container_name: str
+    ) -> bool:
+    """
+    Inserts a new version of a pipeline JSON into an Azure Blob Storage
+    container.
+
+    Args:
+        pipelines_json (dict): The JSON data of the pipeline. connection_string
+        (str): The connection string for the Azure Blob Storage account.
+        pipeline_container_name (str): The name of the container where the
+        pipeline JSON will be uploaded.
+
+    Returns:
+        bool: True if the pipeline JSON was successfully uploaded, False
+        otherwise.
+    """
+    try:
+        blob_service_client = BlobServiceClient.from_connection_string(
```

Comment:
create the clients in main and call your functions with the client as parameter

Reply:
I should probably delete this file since the pipeline insertion work was part of another issue.
* [#60](https://github.com/ai-cfia/nachet-backend/issues/60)

I can implement the change ask on the related PR: [#65](https://github.com/ai-cfia/nachet-backend/pull/65/files)